storage/store.go

```diff
@@ -1075,8 +1086,13 @@ func (s *Store) ProcessRangeDescriptorUpdate(rng *Range) error {
 	}
 	delete(s.uninitRanges, rng.Desc().RaftID)
 
-	s.rangesByKey = append(s.rangesByKey, rng)
-	sort.Sort(s.rangesByKey)
+	if s.rangesByKey.Has((*rangeBTreeItem)(rng)) {
+		return &rangeAlreadyExists{rng}
+	}
+	if exRngItem := s.rangesByKey.ReplaceOrInsert((*rangeBTreeItem)(rng)); exRngItem != nil {
```

Comment:
I don't get it - if we're inserting a range that identical to an existing one (and the existing one is removed) what is the harm?